lib/Hakyll/Web/Feed.hs

```diff
@@ -77,15 +79,15 @@ data FeedConfiguration = FeedConfiguration
 
 --------------------------------------------------------------------------------
 -- | Abstract function to render any feed.
-renderFeed :: String                  -- ^ Default feed template
-           -> String                  -- ^ Default item template
+renderFeed :: Item String             -- ^ Default feed template
+           -> Item String             -- ^ Default item template
```

Comment:
This is my remaining nitpick about the PR -- changing this breaks backwards-compatibility, which is fine, since this is a relatively rarely used function so it's worth it.  However, if we do change it, we should change it to the "right" type which is `Template` or `Item Template` rather than `Item String`.

Reply:
`Template` is fine, but it shouldn't be a template constructed by `readTemplate` which doesn't know about the origin of the template file (it just assumes a string literal), which is important for error messages. Doing the `compileTemplateItem` did allow me to use the `Item`s path for the template, which I had explicitly added, without having to use any of the `Template.Internal` stuff to construct this "embedded `Template`".

Reply:
@bergus That makes a lot of sense -- what do you think of just providing `embedTemplate` from the template modules then?  It has the filepath that way, and simplifies the Feed code: https://github.com/jaspervdj/hakyll/pull/722/commits/f96b99e963b170ab15fe99341cdc7b2b9543dac4